diff -ur ruby-gnome2-all-3.2.5/atk/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/atk/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/atk/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/atk/dependency-check/Rakefile	2018-06-04 08:52:43.257241554 +0200
@@ -15,30 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("atk")
-      unless NativePackageInstaller.install(:altlinux => "libatk-devel",
-                                            :debian => "libatk1.0-dev",
-                                            :redhat => "pkgconfig(atk)",
-                                            :arch => "atk",
-                                            :homebrew => "atk",
-                                            :macports => "atk",
-                                            :msys2 => "atk")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/clutter/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/clutter/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/clutter/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/clutter/dependency-check/Rakefile	2018-06-04 08:57:35.373692720 +0200
@@ -15,27 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("clutter-1.0")
-      unless NativePackageInstaller.install(:debian => "libclutter-1.0-dev",
-                                            :redhat => "pkgconfig(clutter-1.0)",
-                                            :homebrew => "clutter",
-                                            :msys2 => "clutter")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/clutter-gstreamer/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/clutter-gstreamer/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/clutter-gstreamer/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/clutter-gstreamer/dependency-check/Rakefile	2018-06-04 08:55:32.939009990 +0200
@@ -15,27 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("clutter-gst-3.0")
-      unless NativePackageInstaller.install(:debian => "libclutter-gst-3.0-dev",
-                                            :redhat => "pkgconfig(clutter-gst-3.0)",
-                                            :homebrew => "clutter-gst",
-                                            :msys2 => "clutter-gst")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/clutter-gtk/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/clutter-gtk/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/clutter-gtk/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/clutter-gtk/dependency-check/Rakefile	2018-06-04 08:56:21.062867385 +0200
@@ -15,27 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("clutter-gtk-1.0")
-      unless NativePackageInstaller.install(:debian => "libclutter-gtk-1.0-dev",
-                                            :redhat => "pkgconfig(clutter-gtk-1.0)",
-                                            :homebrew => "clutter-gtk",
-                                            :msys2 => "clutter-gtk")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/gdk3/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/gdk3/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/gdk3/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/gdk3/dependency-check/Rakefile	2018-06-04 08:55:44.519135290 +0200
@@ -15,29 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("gdk-3.0")
-      unless NativePackageInstaller.install(:altlinux => "libgtk+3-devel",
-                                            :debian => "libgtk-3-dev",
-                                            :redhat => "pkgconfig(gdk-3.0)",
-                                            :homebrew => "gtk+3",
-                                            :macports => "gtk3",
-                                            :msys2 => "gtk3")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/gdk_pixbuf2/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/gdk_pixbuf2/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/gdk_pixbuf2/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/gdk_pixbuf2/dependency-check/Rakefile	2018-06-04 08:58:14.584135303 +0200
@@ -15,30 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("gdk-pixbuf-2.0")
-      unless NativePackageInstaller.install(:altlinux => "gdk-pixbuf-devel",
-                                            :debian => "libgdk-pixbuf2.0-dev",
-                                            :redhat => "pkgconfig(gdk-pixbuf-2.0)",
-                                            :arch => "gdk-pixbuf2",
-                                            :homebrew => "gdk-pixbuf",
-                                            :macports => "gdk-pixbuf2",
-                                            :msys2 => "gdk-pixbuf2")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/glib2/lib/mkmf-gnome2.rb ruby-gnome2-all-3.2.5-no-native/glib2/lib/mkmf-gnome2.rb
--- ruby-gnome2-all-3.2.5/glib2/lib/mkmf-gnome2.rb	2018-06-04 08:58:30.354314574 +0200
+++ ruby-gnome2-all-3.2.5-no-native/glib2/lib/mkmf-gnome2.rb	2018-06-04 08:54:28.294987267 +0200
@@ -13,7 +13,6 @@
 require 'mkmf'
 require 'pkg-config'
 require 'glib-mkenums'
-require "native-package-installer"
 
 $CFLAGS += " #{ENV['CFLAGS']}" if ENV['CFLAGS']
 
@@ -147,18 +146,8 @@
   end
 end
 
-def setup_homebrew_libffi
-  platform = NativePackageInstaller::Platform.detect
-  return unless platform.is_a?(NativePackageInstaller::Platform::Homebrew)
-
-  libffi_prefix = `brew --prefix libffi`.chomp
-  PKGConfig.add_path("#{libffi_prefix}/lib/pkgconfig")
-end
-
 #add_depend_package("glib2", "ext/glib2", "/...../ruby-gnome2")
 def add_depend_package(target_name, target_srcdir, top_srcdir, options={})
-  setup_homebrew_libffi if target_name == "gobject-introspection"
-
   gem_spec = find_gem_spec(target_name)
   if gem_spec
     target_source_dir = File.join(gem_spec.full_gem_path, "ext/#{target_name}")
@@ -462,25 +451,14 @@
   PKGConfig.have_package('cairo') and have_header('rb_cairo.h')
 end
 
-def install_missing_native_package(native_package_info)
-  NativePackageInstaller.install(native_package_info)
-end
-
 def required_pkg_config_package(package_info, native_package_info=nil)
   if package_info.is_a?(Array)
     required_package_info = package_info
   else
     required_package_info = [package_info]
   end
-  if required_package_info.include?("gobject-introspection-1.0")
-    setup_homebrew_libffi
-  end
   return true if PKGConfig.have_package(*required_package_info)
-
-  native_package_info ||= {}
-  return false unless install_missing_native_package(native_package_info)
-
-  PKGConfig.have_package(*required_package_info)
+  return false
 end
 
 add_include_path = Proc.new do |dir_variable|
diff -ur ruby-gnome2-all-3.2.5/glib2/Rakefile ruby-gnome2-all-3.2.5-no-native/glib2/Rakefile
--- ruby-gnome2-all-3.2.5/glib2/Rakefile	2018-03-31 16:11:24.000000000 +0200
+++ ruby-gnome2-all-3.2.5-no-native/glib2/Rakefile	2018-06-04 08:54:41.835130190 +0200
@@ -38,7 +38,6 @@
   package.description = "Ruby/GLib2 is a Ruby binding of GLib-2.x."
   package.dependency.gem.runtime = [
     ["pkg-config", ">= 1.2.2"],
-    ["native-package-installer", ">= 1.0.3"],
   ]
   package.dependency.gem.development = [["test-unit", ">= 2"]]
   package.windows.packages = []
diff -ur ruby-gnome2-all-3.2.5/gnumeric/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/gnumeric/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/gnumeric/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/gnumeric/dependency-check/Rakefile	2018-06-04 08:54:54.651932882 +0200
@@ -15,30 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    require "gobject-introspection"
-    repository = GObjectIntrospection::Repository.default
-    begin
-      repository.require("Gnm")
-    rescue GObjectIntrospection::RepositoryError::TypelibNotFound
-      unless NativePackageInstaller.install(:arch_linux => "gnumeric",
-                                            :debian => "gir1.2-gnumeric",
-                                            :redhat => "pkgconfig(libspreadsheet-1.12)")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/goffice/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/goffice/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/goffice/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/goffice/dependency-check/Rakefile	2018-06-04 08:57:12.130099264 +0200
@@ -31,38 +31,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("libgoffice-0.10")
-      goffice010_spec = {
-        :debian => "libgoffice-0.10-dev",
-        :redhat => "pkgconfig(libgoffice-0.10)",
-        :homebrew => "goffice",
-      }
-      unless NativePackageInstaller.install(goffice010_spec)
-        unless PKGConfig.check_version?("libgoffice-0.8")
-          goffice08_spec = {
-            :debian => "libgoffice-0.8-dev",
-            :redhat => "pkgconfig(libgoffice-0.8)",
-            :homebrew => "goffice",
-          }
-          unless NativePackageInstaller.install(goffice08_spec)
-            exit(false)
-          end
-        end
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/gsf/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/gsf/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/gsf/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/gsf/dependency-check/Rakefile	2018-06-04 08:56:09.299405149 +0200
@@ -15,27 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("libgsf-1")
-      unless NativePackageInstaller.install(:debian => "libgsf-1-dev",
-                                            :redhat => "pkgconfig(libgsf-1)",
-                                            :homebrew => "libgsf",
-                                            :msys2 => "libgsf")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/gtksourceview3/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/gtksourceview3/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/gtksourceview3/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/gtksourceview3/dependency-check/Rakefile	2018-06-04 08:57:58.683955279 +0200
@@ -15,29 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("gtksourceview-3.0")
-      unless NativePackageInstaller.install(:altlinux => "libgtksourceview3-devel",
-                                            :debian => "libgtksourceview-3.0-dev",
-                                            :redhat => "pkgconfig(gtksourceview-3.0)",
-                                            :homebrew => "gtksourceview3",
-                                            :macports => "gtksourceview3",
-                                            :msys2 => "gtksourceview3")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/poppler/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/poppler/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/poppler/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/poppler/dependency-check/Rakefile	2018-06-04 08:57:23.106888559 +0200
@@ -15,30 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("poppler", 0, 12, 0)
-      unless NativePackageInstaller.install(:alt_linux => "libpoppler-glib-devel",
-                                            :debian => "libpoppler-glib-dev",
-                                            :redhat => "pkgconfig(poppler-glib)",
-                                            :arch_linux => "poppler",
-                                            :homebrew => "poppler",
-                                            :macports => "poppler",
-                                            :msys2 => "poppler")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/rsvg2/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/rsvg2/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/rsvg2/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/rsvg2/dependency-check/Rakefile	2018-06-04 08:56:45.926474793 +0200
@@ -15,30 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("librsvg-2.0")
-      unless NativePackageInstaller.install(:altlinux => "librsvg-devel",
-                                            :debian => "librsvg2-dev",
-                                            :redhat => "pkgconfig(librsvg-2.0)",
-                                            :arch => "librsvg",
-                                            :homebrew => "librsvg",
-                                            :macports => "librsvg",
-                                            :msys2 => "librsvg")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/vte3/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/vte3/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/vte3/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/vte3/dependency-check/Rakefile	2018-06-04 08:55:17.315508447 +0200
@@ -15,28 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-case RUBY_PLATFORM
-when /mingw|mswin/
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("vte-2.91")
-      unless NativePackageInstaller.install(:altlinux => "libvte3-devel",
-                                            :debian => "libvte-2.91-dev",
-                                            :redhat => "pkgconfig(vte-2.91)",
-                                            :homebrew => "vte3")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/webkit2-gtk/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/webkit2-gtk/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/webkit2-gtk/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/webkit2-gtk/dependency-check/Rakefile	2018-06-04 08:56:35.499692927 +0200
@@ -15,26 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("webkit2gtk-4.0")
-      unless NativePackageInstaller.install(:debian => "libwebkit2gtk-4.0-dev",
-                                            :redhat => "pkgconfig(webkit2gtk-4.0)",
-                                            :homebrew => "webkitgtk")
-        exit(false)
-      end
-    end
-  end
-end
diff -ur ruby-gnome2-all-3.2.5/webkit-gtk/dependency-check/Rakefile ruby-gnome2-all-3.2.5-no-native/webkit-gtk/dependency-check/Rakefile
--- ruby-gnome2-all-3.2.5/webkit-gtk/dependency-check/Rakefile	2018-03-14 15:36:59.000000000 +0100
+++ ruby-gnome2-all-3.2.5-no-native/webkit-gtk/dependency-check/Rakefile	2018-06-04 08:55:55.695923388 +0200
@@ -15,26 +15,8 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 require "pkg-config"
-require "native-package-installer"
 
-if File.exist?(File.join(__dir__, "..", "vendor"))
-  task :default => "nothing"
-else
-  task :default => "dependency:check"
-end
+task :default => "nothing"
 
 task :nothing do
 end
-
-namespace :dependency do
-  desc "Check dependency"
-  task :check do
-    unless PKGConfig.check_version?("webkitgtk-3.0")
-      unless NativePackageInstaller.install(:debian => "libwebkitgtk-3.0-dev",
-                                            :redhat => "pkgconfig(webkitgtk-3.0)",
-                                            :msys2 => "webkitgtk3")
-        exit(false)
-      end
-    end
-  end
-end
