diff -ur ruby-gnome2-all-0.10.0-orig/gtksourceview/extconf.rb ruby-gnome2-all-0.10.0/gtksourceview/extconf.rb
--- ruby-gnome2-all-0.10.0-orig/gtksourceview/extconf.rb	2004-08-15 14:26:17.005933000 +0000
+++ ruby-gnome2-all-0.10.0/gtksourceview/extconf.rb	2004-08-15 14:28:23.095188753 +0000
@@ -2,7 +2,7 @@
 extconf.rb for GtkSourceView extention library
 =end
 
-$LOAD_PATH.unshift File.expand_path(File.dirname(__FILE__) + '/..')
+$LOAD_PATH.unshift File.expand_path(File.dirname(__FILE__) + '/../glib/src/lib')
 require 'mkmf'
 require 'mkmf-gnome2'
 
Only in ruby-gnome2-all-0.10.0/gtksourceview: extconf.rb~
