diff -ur ruby-gnome2-all-0.10.0-orig/glib/extconf.rb ruby-gnome2-all-0.10.0/glib/extconf.rb
--- ruby-gnome2-all-0.10.0-orig/glib/extconf.rb	2004-08-08 11:14:49.000000000 -0600
+++ ruby-gnome2-all-0.10.0/glib/extconf.rb	2004-08-08 17:13:49.318608286 -0600
@@ -7,6 +7,7 @@
 require 'mkmf-gnome2'
 
 PKGConfig.have_package('gobject-2.0') or exit 1
+PKGConfig.have_package('glib-2.0') or exit 1
 check_win32
 
 set_output_lib('libruby-glib2.a')
diff -ur ruby-gnome2-all-0.10.0-orig/gtksourceview/extconf.rb ruby-gnome2-all-0.10.0/gtksourceview/extconf.rb
--- ruby-gnome2-all-0.10.0-orig/gtksourceview/extconf.rb	2004-08-08 11:14:49.000000000 -0600
+++ ruby-gnome2-all-0.10.0/gtksourceview/extconf.rb	2004-08-08 17:14:04.834805454 -0600
@@ -14,7 +14,7 @@
 check_win32
 
 top = File.expand_path(File.dirname(__FILE__) + '/..')
-$CFLAGS += " " + ['glib/src'].map{|d|
+$CFLAGS += " " + ['glib/src', 'gtk/src'].map{|d|
   "-I" + File.join(top, d)
 }.join(" ")
 
Only in ruby-gnome2-all-0.10.0/gtksourceview: extconf.rb~
